src/coreclr/vm/stubmgr.cpp

```diff
@@ -2211,6 +2246,15 @@ BOOL TailCallStubManager::TraceManager(Thread * pThread,
     TADDR esp = GetSP(pContext);
     TADDR ebp = GetFP(pContext);
 
+#if defined(TARGET_ARM64) && defined(__APPLE__)
```

Comment:
This code is already inside an `#if defined(TARGET_X86)` starting at line 2195. TARGET_X86 and TARGET_ARM64 shouldn't ever be defined at the same time so this code should never get compiled. I'd recommend removing this part.